data_collection/gazette/spiders/pr_cianorte.py

```diff
@@ -0,0 +1,34 @@
+from dateparser import parse
+from scrapy.http import Request
+
+from gazette.items import Gazette
+from gazette.spiders.base import BaseGazetteSpider
+
+
+class PrCianorteSpider(BaseGazetteSpider):
+    TERRITORY_ID = "4105508"
+    name = "pr_cianorte"
+    allowed_domains = ["cianorte.pr.gov.br"]
+    start_urls = ["https://cianorte.pr.gov.br/orgao-oficial/"]
+    current_page = 1
+    base_url = "https://cianorte.pr.gov.br/orgao-oficial/pagina/"
+
+    def parse(self, response):
+        lines = response.css("tr")[1:]
+        if not lines:
```

Comment:
**suggestion**: You don't need to return if no lines are available. If the list is empty, the `for` loop next will just not be executed and the method will return.

Reply:
@rennerocha is totally right. I'm guessing you didn't find right because of the way the pagination is being made irregardless of having of not having `lines`.

Luckily, we can fix that with a [for-else](https://book.pythontips.com/en/latest/for_-_else.html#for-else). Just putting the request inside the `else` clause would make it work.

Also, a really minor thing that is totally just my preference in thinking it fits nicely: if we do it like this, we could change the `return` statement inside the loop to a `break`.